### PR TITLE
Refactor simple impeller example

### PR DESCRIPTION
**Pull request by jeannejoachim**
# Description of the problem

This example was difficult to read in its formatting, plus there were many information that belonged more in the documentation section.

# Description of the solution

- Reformat the example thoroughly
- Change the figure and add dimensions, consistently with the soon-to-be-completed Nitsche article
- Remove unnecessary parameters (default values)
- Add relevant information in the documentation page
- Add Mesh Adaptation Control section
- Corrected some mistakes in the documentation page
- Added hyperlinks for the prm subsections
- Fixed the embedded videos

# Comments

I suggest to pull the branch and inspect the html files in your browser. You can easily compare it with the previous version (current documentation) like that.

### CODE REVIEW

**Review by blaisb — APPROVED** [review]
Very good, no comments on my side. It's cleaner this way for sure.

File: doc/source/parameters/cfd/nitsche.rst [hunk 1]
```diff
@@ -51,7 +58,7 @@ These parameters are used for simulations using the Nitsche immersed boundary me
       # Force and torque calculation on solid
       set calculate force on solid = false
       set solid force name = force_solid
-      set calculate torque on solid = true
+      set calculate torque on solid = false
```
Comment (CaroleAnneDaunais) [inline doc/source/parameters/cfd/nitsche.rst:61]: I guess you can remove all the parameters that are false by default!

**Review by CaroleAnneDaunais — APPROVED** [review]
Nice modifications!

File: doc/source/parameters/cfd/nitsche.rst (hunk 1 above)
Comment (jeannejoachim) [reply doc/source/parameters/cfd/nitsche.rst:61]: This is in the documentation, not in the example :) I saw this error while I was checking the parameters of the example (I removed the parameters that are used at their default values).

File: doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitsche_mixer_with_pbt_impeller.rst [hunk 2]
```diff
@@ -180,54 +254,85 @@ It is often desirable to obtain the time-averaged velocity field when simulating
...
 
 .. code-block:: text
 
-  gls_nitsche_navier_stokes_33 mixer.prm
+  mpirun -np <np> </path/to/the/build/applications/gls_nitsche_navier_stokes_33/gls_nitsche_navier_stokes_33 mixer.prm>
 
-Lethe generates a larger number of files when the Nitsche immersed boundary is used. The ``mixer.pvd`` contains the 3D visualization of the velocity, pressure, and auxiliary variables such as the average velocity. Additional files will also be generated, notably the ``mixer_solid_triangulation_00.pvd`` and ``mixer_solid_particles_00.pvd``, corresponding to the Nitsche IB mesh and the discrete particles inserted at the Gauss points of the solid triangulation, respectively. The solid particles enable the Nitsche restriction visualization, while the solid triangulation is used for animation purposes.
+Lethe generates a larger number of files when the Nitsche immersed boundary is used:
+* the ``mixer.pvd`` contains the 3D visualization of the velocity, pressure, and auxiliary variables such as the average velocity ;
```
Comment (bibeauv) [inline doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitsche_mixer_with_pbt_impeller.rst:328]: The bullets does not appear on the html. I think it needs an indentation?

**Review by bibeauv — COMMENTED** [review]
Really well written and super super clear :)

File: doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitsche_mixer_with_pbt_impeller.rst [hunk 3]
```diff
@@ -180,54 +254,85 @@ It is often desirable to obtain the time-averaged velocity field when simulating
...
-    set amg smoother sweeps       = 2      
-    set amg smoother overlap      = 1      
-    set verbosity                 = verbose
-    set max krylov vectors      = 200
+    set max krylov vectors        = 200
   end
 
+:raw-html:`<br />`
+
 Running the simulation
```
Comment (AmishgaAlphonius) [inline doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitsche_mixer_with_pbt_impeller.rst:319]:
```suggestion
Running the Simulation
```

File: doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitsche_mixer_with_pbt_impeller.rst [hunk 4]
```diff
@@ -267,6 +374,7 @@ From these animations, we can see that the impeller generates an axial pumping f
 Clearly, the fluid is descending along the shaft and rises along the lateral walls of the vessel.
 Multiple analyses can be done with this type of simulation. For example, one can monitor the torque on the impeller as a function of the Reynolds number to generate the power curve of the agitator. One could also leverage the tracer multiphysics module to measure the mixing time of the tank and/or identify the presence of dead zones.
 
+:raw-html:`<br />`
 
 Possibilities for extension
```
Comment (AmishgaAlphonius) [inline doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitsche_mixer_with_pbt_impeller.rst:379]:
```suggestion
Possibilities for Extension
```

**Review by AmishgaAlphonius — APPROVED** [review]
Only the 2 last titles weren't capitalized ;)

File: doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitsche_mixer_with_pbt_impeller.rst [hunk 5]
```diff
@@ -180,54 +254,85 @@ It is often desirable to obtain the time-averaged velocity field when simulating
...
 
 .. code-block:: text
 
-  gls_nitsche_navier_stokes_33 mixer.prm
+  mpirun -np <np> </path/to/the/build/applications/gls_nitsche_navier_stokes_33/gls_nitsche_navier_stokes_33 mixer.prm>
 
-Lethe generates a larger number of files when the Nitsche immersed boundary is used. The ``mixer.pvd`` contains the 3D visualization of the velocity, pressure, and auxiliary variables such as the average velocity. Additional files will also be generated, notably the ``mixer_solid_triangulation_00.pvd`` and ``mixer_solid_particles_00.pvd``, corresponding to the Nitsche IB mesh and the discrete particles inserted at the Gauss points of the solid triangulation, respectively. The solid particles enable the Nitsche restriction visualization, while the solid triangulation is used for animation purposes.
+Lethe generates a larger number of files when the Nitsche immersed boundary is used:
+* the ``mixer.pvd`` contains the 3D visualization of the velocity, pressure, and auxiliary variables such as the average velocity ;
```
Comment (jeannejoachim) [inline doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitsche_mixer_with_pbt_impeller.rst:328]:
```suggestion

* the ``mixer.pvd`` contains the 3D visualization of the velocity, pressure, and auxiliary variables such as the average velocity ;
```

File: doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitsche_mixer_with_pbt_impeller.rst (hunk 2 above)
Comment (jeannejoachim) [reply doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitsche_mixer_with_pbt_impeller.rst:328]: Well spotted! I need to add a line between the text and bullet point

**Pull request merged**